test/e2e/aws_test.go

```diff
@@ -118,6 +140,172 @@ var _ = Describe("functional tests", func() {
 	})
 })
 
+func scaleMachineDeployment(namespace, machineDeployment string, replicasCurrent int32, replicasProposed int32) {
+	fmt.Fprintf(GinkgoWriter, "Scaling MachineDeployment from %d to %d\n", replicasCurrent, replicasProposed)
+	deployment := &clusterv1.MachineDeployment{}
+	Expect(kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: machineDeployment}, deployment))
+	deployment.Spec.Replicas = &replicasProposed
+	Expect(kindClient.Update(context.TODO(), deployment)).NotTo(HaveOccurred())
+	waitForMachinesCountMatch(namespace, machineDeployment, replicasCurrent, replicasProposed)
+	waitForMachineDeploymentRunning(namespace, machineDeployment, replicasProposed)
```

Comment:
It might be okay to just skip `waitForMachinesCountMatch` and just use `waitForMachineDeploymentRunning`

Reply:
In case of scaling down the MachineDeployment, Replicas and ReadyReplicas will be set to replica value immediately after marking the machines for deletion. But the actual deletion will be happening in the back ground. So, the waitForMachineDeploymentRunning() method call will get executed quickly and the next operation of scale UP/DOWN will be delayed as the previous tasks were not completed. Hence, i wanted to ensure that scale down is completed in terms of machines.